core/workspace.js

```diff
@@ -523,8 +523,8 @@ Blockly.Workspace.prototype.fireChangeListener = function(event) {
   if (event.recordUndo) {
     this.undoStack_.push(event);
     this.redoStack_.length = 0;
-    if (this.undoStack_.length > this.MAX_UNDO) {
-      this.undoStack_.unshift();
+    while (this.undoStack_.length > this.MAX_UNDO) {
+      this.undoStack_.shift();
```

Comment:
Not for this PR, but we may want to improve this later to size it by the number of undo groups instead of individual events.

Reply:
As far as I know, in the real world this setting will be either zero, or a "big number".  The "big number" is just a limit to prevent swamping memory.  I can't think of a use-case where someone would actually want the number of undo operations to be a specific number such as 6.  In the "big number" case, what we actually care about is bytes, not undo events or undo groups.

Thus I don't think that counting groups would help anyone.